src/dotnet-new3/Program.cs

```diff
@@ -69,14 +69,13 @@ private static DefaultTemplateEngineHost CreateHost(bool emitTimings, bool disab
             { }
 
             // Keep this in sync with dotnet/sdk repo
-            var builtIns = new List<KeyValuePair<Guid, Func<Type>>>(new AssemblyComponentCatalog(new[]
-            {
-                typeof(Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Abstractions.IMacro).GetTypeInfo().Assembly
-            }));
-
+            var builtIns = new List<(Type, IIdentifiedComponent)>();
+            builtIns.AddRange(Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Components.AllComponents);
+            builtIns.AddRange(Microsoft.TemplateEngine.Edge.Components.AllComponents);
```

Comment:
should we rather keep it in Edge/CLI? but via AddComponent()? I like the idea they can register themselves, especially Edge.

Reply:
Benefit of having this is to allow host to disable GlobalSettingsTemplatePackageProviderFactory if they wish so...

Reply:
Also it feels more in control and all in one place instead of some self adding magic...

Reply:
On the opposite side, each host needs to know to add these Edge components... It makes it difficult for simple hosts scenarios. Anyway we have https://github.com/dotnet/templating/issues/2613 - I will add default registration there after this is merged.